odbc/src/api/data.rs

```diff
@@ -140,29 +142,89 @@ impl IntoCData for Bson {
         }
     }
 
-    fn to_u64(&self) -> Result<(u64, Option<ODBCError>)> {
-        self.to_i64().map(|(i, w)| (i as u64, w))
-    }
-
     fn to_i32(&self) -> Result<(i32, Option<ODBCError>)> {
-        let (out_64, warning) = self.to_i64().map_err(|e| match e {
+        self.to_i64().map_err(|e| match e {
             ODBCError::RestrictedDataType(s, _) => ODBCError::RestrictedDataType(s, INT32),
             ODBCError::InvalidCharacterValue(s, _) => ODBCError::InvalidCharacterValue(s, INT32),
             _ => e,
         })?;
-        let out = out_64 as i32;
-        Ok((
-            out,
-            warning.or(if out as i64 != out_64 {
-                Some(ODBCError::IntegralTruncation(self.to_string()))
-            } else {
-                None
-            }),
-        ))
+        match self {
+            Bson::Double(x) if *x > i32::MAX as f64 => {
```

Comment:
is the inverse of this possible (e.g. `*x < i32::MIN`)? What would be the expected behavior in that case?

Reply:
That's a great point @jchemburkar. Good catch!